tests/fetch/response_headers.cpp

```diff
@@ -0,0 +1,78 @@
+#include <string.h>
+#include <stdio.h>
+#include <math.h>
+#include <assert.h>
+#include <emscripten/fetch.h>
+
+int result = 0;
+
+int main()
+{
+    static const char* const headers[] = {
+        "X-Emscripten-Test",
+        "1",
+        0,
+    };
+    const size_t n_values = sizeof( headers ) / sizeof( headers[ 0 ] ) - 1;
+    emscripten_fetch_attr_t attr;
+    emscripten_fetch_attr_init( &attr );
+    strcpy( attr.requestMethod, "GET" );
+    attr.attributes = EMSCRIPTEN_FETCH_REPLACE | EMSCRIPTEN_FETCH_LOAD_TO_MEMORY | EMSCRIPTEN_FETCH_SYNCHRONOUS;
+    attr.requestHeaders = headers;
+
+    attr.onsuccess = [] ( emscripten_fetch_t *fetch )
+    {
+        assert( fetch->__attributes.requestHeaders != 0 );
+        assert( fetch->__attributes.requestHeaders != headers );
+        for ( size_t i = 0; i < n_values; ++i )
+        {
+            const char* origHeader = headers[ i ];
+            const char* header = fetch->__attributes.requestHeaders[ i ];
+            assert( origHeader != header );
+            assert( strcmp( origHeader, header ) == 0 );
+        }
+        assert( fetch->__attributes.requestHeaders[ n_values ] == 0 );
+
+        printf( "Finished downloading %llu bytes\n", fetch->numBytes );
+        // Compute rudimentary checksum of data
+        uint8_t checksum = 0;
+        for ( int i = 0; i < fetch->numBytes; ++i )
+            checksum ^= fetch->data[ i ];
+        printf( "Data checksum: %02X\n", checksum );
+        assert( checksum == 0x08 );
+        emscripten_fetch_close( fetch );
+
+        if ( result == 0 ) result = 1;
+#ifdef REPORT_RESULT
+        REPORT_RESULT( result );
+#endif
+    };
+
+    attr.onprogress = [] ( emscripten_fetch_t *fetch )
+    {
+        if ( fetch->totalBytes > 0 )
+        {
+            printf( "Downloading.. %.2f%% complete.\n", ( fetch->dataOffset + fetch->numBytes ) * 100.0 / fetch->totalBytes );
+        }
+        else
+        {
+            printf( "Downloading.. %lld bytes complete.\n", fetch->dataOffset + fetch->numBytes );
+        }
+    };
+
+    attr.onerror = [] ( emscripten_fetch_t *fetch )
+    {
+        printf( "Download failed!\n" );
+        assert( false && "Shouldn't fail!" );
```

Comment:
The `onerror` handler is good to have a call to `emscripten_fetch_close(fetch);` as well. Even though that should not happen in this test, occassionally people copy paste test code to use as their own basis, so it's good to have this follow the good practice.